.github/workflows/build-publish.yaml

```diff
@@ -3,113 +3,49 @@ name: Build and Publish Docker image
-          else
-            echo "::set-output name=tag::${GITHUB_REF##*/}-$CI_COMMIT_SHORT_SHA"
-          fi
-        env:
-          version_pattern: "tags\\/v[0-9]+\\.[0-9]+\\.[0-9]+"
-  build:
-    runs-on: ubuntu-latest
-    needs: prepare
-    environment:
-        name: cloudops
+      - name: Check out the repo
+        uses: actions/checkout@v3
 
-    steps:
-      - name: Docker image tag
-        env:
-          IMAGE_TAG: ${{ needs.prepare.outputs.FULL_IMAGE_TAG }}
+      - name: Build `version.json` file
         run: |
-          echo "Building an image with the following tag:"
-          echo $IMAGE_TAG
-
-      - name: Checkout
-        uses: actions/checkout@v2
-
-      - name: Login to Docker Hub
-        uses: docker/login-action@v1
-        with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
-
-      - name: Extract metadata (tags, labels) for Docker
-        uses: docker/metadata-action@v3
-        with:
-          images: $GITHUB_REPOSITORY
-
-      - name: Build Docker image
-        env:
-          IMAGE_TAG: ${{ needs.prepare.outputs.FULL_IMAGE_TAG }}
-        run: |
-          printf '{\n    "commit": "%s",\n    "version": "%s",\n    "image_tag": "%s",\n    "source": "%s",\n    "build": "%s"\n}\n' \
+          printf '{\n    "commit": "%s",\n    "version": "%s",\n    "source": "%s",\n    "build": "%s"\n}\n' \
             "$GITHUB_SHA" \
             "$GITHUB_REF" \
-            "$IMAGE_TAG" \
             "$GITHUB_SERVER_URL/$GITHUB_REPOSITORY" \
             "$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID" > ./version.json
-          docker build --file Dockerfile -t $GITHUB_REPOSITORY:$IMAGE_TAG .
-
-      - name: Save image as artifact
-        env:
-          IMAGE_TAG: ${{ needs.prepare.outputs.FULL_IMAGE_TAG }}
-        run: |
-          docker save -o /tmp/container.tar.gz $GITHUB_REPOSITORY:$IMAGE_TAG
-
-      - uses: actions/upload-artifact@v3
-        with:
-          name: container-image
-          path: /tmp/container.tar.gz
-          retention-days: 1  # Instead of default 90 days
-
-  publish:
-    runs-on: ubuntu-latest
-    needs:
-      - prepare
-      - build
-    if: ${{ !github.event.pull_request }} # Exclude pull-requests (see `on:` above)
 
-    steps:
-      - uses: actions/download-artifact@v3
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v4
         with:
-          name: container-image
-          path: /tmp/container.tar.gz
-
-      - name: Load container from artifact
-        run: |
-          docker load -i /tmp/container.tar.gz
+          images: $GITHUB_REPOSITORY
+          # https://github.com/marketplace/actions/docker-metadata-action#tags-input
+          tags: |
+            type=semver,pattern={{version}}
+            type=raw,value=latest,enable={{is_default_branch}}
+            type=sha,enable={{is_default_branch}}
```

Comment:
You'll see in the first action run for this PR that there were a warnings:
```
Warning: No Docker image version has been generated. Check tags input.
Warning: No Docker tag has been generated. Check tags input.
```

I'm not _exactly_ sure about this, but my thinking is that since none of these rules apply (this wasn't a push to `main` nor a tagged commit), tags and and a version weren't generated. Maybe we should remove the `enabled={{is_default_branch}}` and `github.event_name != 'pull_request'` for a run or two to test that this works? Or YOLO and see what happens after a merge